File: coding_test_4/quiz2.js
function solution(m, v) {
    let answer = [];

    for(let vv of v){
        let index = answer.findIndex(a => (m - a) < vv)
        // for(let i in answer){
        //     if(m - answer[i] >= vv)
        //         index = i
        //     else break
        // }

        if(index > 0)
            answer[index - 1] += vv
        else if(m - answer[0] >= vv){
            answer[0] += vv
        }else {
            answer.unshift(vv)
        }
    }

    return answer.length;
}


console.log(
    solution(4, [2,3,1])
)

File: coding_test_3/quiz2.py
def find_w_in_line(card, word):
    visited = [False] * 3

    for c in word:
        found = False

        for i in range(3):
            if c in card[i]:
                visited[i] = True
                found = True
                card[i] = card[i].replace(c, '', 1)

        if not found:
            return False

    return all(visited)


def solution(card, word):
    answer = []

    for w in word:
        if find_w_in_line(card[:], w):
            answer.append(w)

    return answer if len(answer) > 0 else ["-1"]

print(
    solution(
        ["ABACDEFG", "NOPQRSTU", "HIJKLKMM"],
        ["GPQM", "GPMZ", "EFU", "MMNA"]
    )
)
